FilePersistence/src/version_control/GitRepository.cpp

```diff
@@ -318,30 +318,51 @@ CommitGraph GitRepository::commitGraph(QString startRevision, QString endRevisio
 
 const Commit* GitRepository::getCommit(QString revision) const
 {
-	Q_ASSERT(revision != WORKDIR);
 	Q_ASSERT(revision != INDEX);
+	if (revision == WORKDIR)
+	{
+		auto m = new Commit();
+		QDirIterator DirIter(workdirPath(), QDirIterator::Subdirectories);
+		while (DirIter.hasNext())
+		{
+			DirIter.next();
+			if (DirIter.fileInfo().isFile())
+			{
+				QFile file(DirIter.filePath());	// projects/Testproject
+				if (!file.open(QIODevice::ReadOnly | QIODevice::Text)) continue;
+				QByteArray text = file.readAll();
+				char *content = new char[text.size() + 1];
+				strcpy(content, text.data());
```

Comment:
Actually, please use the more secure version of this function: http://www.cplusplus.com/reference/cstring/strncpy/